src/LibHdb++Cassandra.cpp

```diff
@@ -470,11 +472,26 @@ bool HdbPPCassandra::find_attr_id_and_ttl_in_db(string fqdn_attr_name, CassUuid
 		if(get_ttl_cass_error != CASS_OK)
 		{
 			if (get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
+			{
+				// The ttl was null, perhaps it was removed, or missing due to an
+				// upgrade. This is not an error condition
 				DEBUG_MSG("TTL value is NULL" << endl);
+			}
+			else
+			{
+				cass_result_free(result);
+				cass_iterator_free(iterator);
+				cass_future_free(future);
+				cass_statement_free(statement);
+
+				stringstream error_desc;
 
-			DEBUG_MSG("An error occured: " << cass_error_desc(get_ttl_cass_error) << endl);
-			DEBUG_MSG("cass_value_get_int32 for ttl detected a NULL value" << endl);
-			// TODO (SJ) Is this an error condition?
+				error_desc << "An unexpected database error occured when trying to retrieve the TTL: "
```

Comment:
Could you please add the fqdn_attr_name (name of the attribute) in the error description? This would help to solve the problem.